sqlmesh/dbt/model.py

```diff
@@ -423,6 +439,81 @@ def to_sqlmesh(self, context: DbtContext) -> Model:
                     "target_lag": self.target_lag,
                 }
 
+        if context.target.dialect == "clickhouse":
+            unsupported_materializations = [
+                "dictionary",
+                "distributed_table",
+                "distributed_incremental",
+                "materialized_view",
+            ]
+            if self.materialized in unsupported_materializations:
+                fallback = self.materialized.split("_")
+                msg = f"SQLMesh does not support the '{self.materialized}' model materialization."
+                if len(fallback) == 1:
+                    # dictionary materialization
+                    raise ConfigError(msg)
+                else:
+                    logger.warning(f"{msg} Falling back to the '{fallback[1]}' materialization.")
+                    self.materialized = fallback[1]
```

Comment:
Not sure I like the idea of mutating this object during the conversion.

Reply:
Agreed, but we can't wait until later because we call the `materialization` property that passes the `materialized` string to an enum on the next line.

Should I add a materialized validator and do it there?

Reply:
Validator makes sense to me. Otherwise `model_materialization` is not usable at all with clickhouse